### PR TITLE
Updated installation instructions

### PR DESCRIPTION
**Pull request by homotopycolimit**
- updated installation instructions
- renaming the executable from swarm to go-swarm
- changed version from 0.2 to 0.3

### CODE REVIEW

File: contents/installation.rst [hunk 1]
```diff
@@ -48,52 +56,71 @@ Then you must prepare your go environment as follows
...
+  sudo apt-get install golang-1.10-go
 
-Ubuntu users can use the 'gophers' PPA to install an up to date version of Go (version 1.7 or later is preferred). See https://launchpad.net/~gophers/+archive/ubuntu/archive for more information. Note that this PPA requires adding /usr/lib/go-1.X/bin to the executable PATH.
+and then add ``/usr/lib/go-1.10/bin`` to your ``PATH`` environment variable.
 
-Other distros
+.. code-block:: none
+
+  export PATH="$PATH:/usr/lib/go-1.10/bin"
+  echo 'export PATH="$PATH:/usr/lib/go-1.10/bin"' >> ~/.bashrc
```
Comment (justelad) [inline contents/installation.rst]: i think you can append path with just:
`echo 'export PATH=$PATH:/usr/lib/go-1.10/bin' >> ~/.bashrc`

File: contents/installation.rst [hunk 2]
```diff
@@ -48,52 +56,71 @@ Then you must prepare your go environment as follows
...
+
+You must also set up a go folder and ``GOPATH``.
```
Comment (justelad) [inline contents/installation.rst]:
```
mkdir $HOME/go
echo 'export GOPATH=$HOME/go' >> ~/.bashrc
echo 'export PATH=$PATH:$GOPATH/bin' >> ~/.bashrc
?
```

File: contents/installation.rst [hunk 3]
```diff
@@ -48,52 +56,71 @@ Then you must prepare your go environment as follows
...
 
+.. _below:
 
 Ubuntu
-================
+---------
+
+At the time of writing, the Ubuntu repositories carry an older version of Go. Up to date instruction on how to install the newest version of Go in Ubuntu can always be found `here <https://github.com/golang/go/wiki/Ubuntu>`_.
+
+Ubuntu users can use the 'gophers' PPA to install an up to date version of Go. See https://launchpad.net/~gophers/+archive/ubuntu/archive for more information. Note that this PPA requires adding /usr/lib/go-1.X/bin to the executable PATH.
```
Comment (lmars) [inline contents/installation.rst]: I really don't think we should be telling people how to install Go, we should just defer to the Go documentation for that (which means our docs won't keep getting outdated). This is also [what geth does](https://github.com/ethereum/go-ethereum/wiki/Installing-Geth#build-it-from-source-code).

File: contents/installation.rst [hunk 4]
```diff
@@ -48,52 +56,71 @@ Then you must prepare your go environment as follows
...
-Read the environment variables into current session: 
+Read the environment variables into current session:
 
 .. code-block:: none
 
   source ~/.bashrc
 
-Installing from source
-=======================
+Installing Swarm from source
```
Comment (lmars) [inline contents/installation.rst]: Is there a reason we don't encourage people to just download a signed swarm binary from a download page? This is quite an involved process for someone who just wants to get going using Swarm.

File: contents/installation.rst (hunk 2 above)
Comment (homotopycolimit) [reply contents/installation.rst]: I'll go with: (both here and on line 52 above)
  mkdir $HOME/go
  export GOPATH="$HOME/go"
  echo 'export GOPATH=$HOME/go' >> ~/.bashrc
  export PATH=$PATH:$GOPATH/bin
  echo 'export PATH=$PATH:$GOPATH/bin' >> ~/.bashrc

File: contents/installation.rst (hunk 3 above)
Comment (homotopycolimit) [reply contents/installation.rst]: so, just keep the link to the golang wiki and lose the rest?
@lmars

File: contents/installation.rst (hunk 4 above)
Comment (homotopycolimit) [reply contents/installation.rst]: I have never installed from binary. How does it work? Is there a ppa to add? what should be in the docs?

**Comment by homotopycolimit** [comment]
I have incorporated Elad's fixes now.

Lewis' suggestion to add the instructions for installing the binary is valid. We should include that but maybe we make that another PR?

File: contents/installation.rst (hunk 4 above)
Comment (justelad) [reply contents/installation.rst]: i think that in the sense of installing from binary we should just fall back to the binaries shipped with the geth client?

**Comment by justelad** [comment]
@lmars merged this into my branch and pasted your pending comments in the relevant lines in my open PR.
we will address this as part of the PR.
@homotopycolimit fyi

File: contents/installation.rst (hunk 3 above)
Comment (lmars) [reply contents/installation.rst]: I think just do exactly what Geth does (link to generic Go installation / download pages).

File: contents/installation.rst (hunk 4 above)
Comment (lmars) [reply contents/installation.rst]: You can get a swarm binary by downloading the "Geth & Tools" package from https://geth.ethereum.org/downloads/.

We should probably get whoever maintains that to make Swarm a more explicitly mentioned download.

**Pull request merged**